finesse/gui/stepper_motor_view.py

```diff
@@ -0,0 +1,67 @@
+"""Code for controlling the stepper motor which moves the mirror."""
+from typing import Optional
+
+from pubsub import pub
+from PySide6.QtWidgets import QGridLayout, QGroupBox, QPushButton, QSpinBox
+
+
+class StepperMotorControl(QGroupBox):
+    """A control showing buttons for moving the mirror to a target."""
+
+    def __init__(self) -> None:
+        """Create a new StepperMotorControl."""
+        super().__init__("Target control")
+
+        layout = QGridLayout()
+        zenith = self._create_stepper_button("ZENITH")
+        nadir = self._create_stepper_button("NADIR")
+        hot_bb = self._create_stepper_button("HOT_BB")
+        cold_bb = self._create_stepper_button("COLD_BB")
+        home = self._create_stepper_button("HOME")
+        self.angle = QSpinBox()
+        self.angle.setMaximum(359)
+        self.goto = QPushButton("GOTO")
+        self.goto.clicked.connect(self._goto_clicked)  # type: ignore
+
+        self.last_clicked: Optional[QPushButton] = None
+
+        layout.addWidget(zenith, 0, 0)
+        layout.addWidget(nadir, 0, 1)
+        layout.addWidget(hot_bb, 0, 2)
+        layout.addWidget(cold_bb, 0, 3)
+        layout.addWidget(home, 1, 0)
+        layout.addWidget(self.angle, 1, 2)
+        layout.addWidget(self.goto, 1, 3)
+
+        self.setLayout(layout)
+
+    def _create_stepper_button(self, name: str) -> QPushButton:
+        """Create a button to move the motor to a preset position.
+
+        Args:
+            name: The name of the preset (will be converted to lowercase)
+        """
+        btn = QPushButton(name)
+        btn.clicked.connect(lambda: self._preset_clicked(btn))  # type: ignore
+        return btn
+
+    def _set_selected_button(self, btn: QPushButton) -> None:
+        """Change the currently selected button, indicated by colour.
+
+        Args:
+            btn: The just-clicked button
+        """
+        if self.last_clicked:
+            self.last_clicked.setStyleSheet("")
+        btn.setStyleSheet("background-color: blue")
+        self.last_clicked = btn
+
```

Comment:
I'm not totally sure, but the way you're using this makes me think on button groups. Have a look at this in case that is what you actually want to use: https://doc.qt.io/qtforpython/PySide6/QtWidgets/QButtonGroup.html

Reply:
Ahh that's neat! I managed to rewrite my "make selected buttons blue" logic using a `QButtonGroup`, which is much cleaner.